tls/s2n_handshake_io.c

```diff
@@ -483,9 +497,20 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     /* A handshake type has been negotiated */
     conn->handshake.handshake_type = NEGOTIATED;
 
-    /* In the initial TLS1.3 release, we will only support the basic handshake. */
+    s2n_cert_auth_type client_cert_auth_type;
```

Comment:
is there a reason to move this above != FULL_HANDSHAKE?

Reply:
To avoid code duplication. For 1.3 we bail early as we only have full, negotiated, client auth or no cert. 1.2 has other message types so needs to continue. Otherwise, I can move L501-L509 back to where it was and make a copy inside of the IS_TLS13_HANDSHAKE().